File: kakao2018/비밀지도.py
def solution(n, arr1, arr2):
    arr = list(map(lambda a : a[0] | a[1], zip(arr1, arr2)))
    arr = list(map(lambda a : '0'*(n-len(bin(a)[2:])) + bin(a)[2:], arr))
    result = []
    for a in arr :
        a = a.replace('1', '#')
        a = a.replace('0', ' ')
        result.append(a)
    return result

File: kakao2018/비밀지도.js
function solution(n, arr1, arr2) {
    let result = [];
    // 지도(배열) 합치기
    for(let i=0; i<arr1.length; i++) {
        result.push(arr1[i] | arr2[i]);
    }
    
    // 2진수 변환
    let maxLen = Math.max(...result).toString(2).length;
    result = result.map(val => val.toString(2)).map(val => '0'.repeat(maxLen-val.length) + val);
    
    // 암호화
    result = result.map(str => str.split('').map(num => num == 1 ? '#' : ' ').join(''));
    return result;
}
